File: java/stack/StackUsingArray.java
package stack;

import java.util.NoSuchElementException;

public class StackUsingArray {
    private Integer [] stack;
    private Integer size;
    private Integer top;

    public StackUsingArray(int size){
        stack = new Integer[size];
        size =size;
        top = -1;
    }

    public boolean isFull(){
        return top == size-1;
    }

    public boolean isEmpty(){
        return top ==-1;
    }


    public int pop(){
        if(top ==-1) throw new NoSuchElementException("Stack is Empty.");
        int ele = stack[top];
        top--;
        return ele;
    }

    public int peek(){
        if(top ==-1) throw new NoSuchElementException("Stack is Empty");
        return stack[top];
    }

    public void push(Integer element){
        if(this.isFull()) throw new IllegalStateException("Stack is Full");
        top++;
        stack[top] = element;
    }
}

File: python/sandeepk/stack/StackUsingArray.py
class StackUsingArray:

    def __init__(self, size):
        self.stack = [size]
        self.size = size
        self.top = -1

    def push(self, data):
        if self.is_full():
            print('Stack is Full')
        else:
            self.top += 1
            self.stack[self.top] = data
            print('Successfully pushed element to stack')

    def pop(self):
        if self.is_empty():
            print('Stack is empty')
        else:
            popped = self.stack[self.top]
            self.top -= 1
            print('Successfully popped element from stack')
            return popped

    def is_full(self):
        return self.top == self.size-1

    def is_empty(self):
        return self.top == -1
